### DISCUSSION TITLE
Accomodating GCP error codes in docs and workflow design.

### DISCUSSION THREAD

**Discussion by r0f1** [nextflow-io/nextflow#1913] [General]
Hi, I am referring to this section: https://www.nextflow.io/docs/latest/google.html#preemptible-instances

Yes it is true, that if a VM gets preemted, the error code is 14, but that is not the whole truth.

According to the Google Life Sciences documentation it can be 8, 10 or 14. See: https://cloud.google.com/life-sciences/docs/troubleshooting, Section: Retrying after encountering errors

It would be nice to add that info to the docs as well.

**Comment by abhi18av** [comment]
Hi @r0f1 , thank you for your suggestion :) 

Since the nature of these error codes is independent of the NF example mentioned in the docs, perhaps it'd be more appropriate to add a `TIP` in the docs to refer the official error code docs from GCP https://cloud.google.com/life-sciences/docs/troubleshooting#error_codes 

This might be safer, since GCP might decide to add / update the error codes in future.

What do you think?

**Comment by r0f1** [comment]
Sure, add it as a `TIP`. Just add it in some way such that developers who want to use preemptible machines know what to look out for. Given that the default value of `google.lifeSciences.preembtible` is `true`, I think this is especially important. Here is what I did by the way, but maybe there is a more elegant way:
```
# nextflow.config
process {
    withLabel: foo {
        executor = "google-lifesciences"
        errorStrategy = {(task.exitStatus==8 || task.exitStatus==10 || task.exitStatus==14) ? 'retry' : 'terminate'}
        maxRetries = 10
    }
}
```

**Reply by abhi18av** [reply]
Yup, that makes sense 👍 

However, I also think that you could probably extract that out as a reusable function like shown [here](https://github.com/nf-core/rnaseq/blob/master/modules/nf-core/software/gffread/functions.nf) .

**Reply by abhi18av** [reply]
@r0f1 , with this https://github.com/nextflow-io/nextflow/pull/1914 we've added a tip regarding the GLS error codes.
